### PR TITLE
[expo-linking][ios] Fix openSettings

### LINKED ISSUE #7700: Add Linking.openSettings native method on iOS
Follow up from https://github.com/expo/expo/pull/7676

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Resolves https://github.com/expo/expo/issues/7700.

# How

- Exported `openSettings` method from the `EXLinkingManager` module.
Please, check if I correctly updated the `react-native` fork. I'm not sure to which branch should I open PR.

# Test Plan

- `Linking.openSettings`(on iOS) ✅

### CODE REVIEW

File: packages/expo/src/Linking/Linking.ts [hunk 1]
```diff
@@ -189,9 +189,4 @@ newLinking.makeUrl = makeUrl;
 newLinking.parse = parse;
 newLinking.parseInitialURLAsync = parseInitialURLAsync;
 
-// TODO: remove this on SDK 38 after adding to EXLinking
-if (Platform.OS === 'ios') {
-  newLinking.openSettings = () => newLinking.openURL('app-settings:');
```
Comment (EvanBacon) [inline packages/expo/src/Linking/Linking.ts:194]: Why would we add more native code when it already works in JS?

Comment (brentvatne) [reply packages/expo/src/Linking/Linking.ts:194]: better to not depend on hardcoded `'app-settings:'` url and instead use `UIApplicationOpenSettingsURLString`

**Comment by brentvatne** [comment]
left a comment on the expo/react-native PR, we don't want to pull this in to sdk-37 but rather ship it with sdk-38

File: packages/expo/src/Linking/Linking.ts (hunk 1 above)
Comment (EvanBacon) [reply packages/expo/src/Linking/Linking.ts:194]: I could see `app-settings:` being used as a URI scheme though, like `app-settings://camera` which could scale to work with Android better.

Comment (brentvatne) [reply packages/expo/src/Linking/Linking.ts:194]: people should use intentlauncher on android for this - https://docs.expo.io/versions/v37.0.0/sdk/intent-launcher/

we could potentially make a higher level api around this but for now the idea is to just make sure we provide the implementation for `Linking.openSettings()` that matches upstream

**Review by brentvatne — APPROVED** [review]

**Pull request merged**